BookLibrary-cdi/src/main/java/org/booklibrary/app/persistence/session/common/AbstractGenericEntityPersistence.java

```diff
@@ -0,0 +1,127 @@
+package org.booklibrary.app.persistence.session.common;
+
+import org.booklibrary.app.persistence.entity.AbstractBaseEntity;
+import org.booklibrary.app.persistence.id.EntityIdentifier;
+import org.booklibrary.app.persistence.session.common.GenericFacadeLocal;
+import org.booklibrary.app.persistence.session.common.GenericHomeLocal;
+import org.slf4j.Logger;
+
+import javax.persistence.EntityManager;
+import java.lang.reflect.ParameterizedType;
+import java.util.List;
+
+/**
+ * Base abstract class for all entity persistence related classes.
+ *
+ * @see org.booklibrary.app.persistence.session.common.GenericHomeLocal
+ * @see org.booklibrary.app.persistence.session.common.GenericFacadeLocal
+ */
+public abstract class AbstractGenericEntityPersistence<T extends AbstractBaseEntity, PK>
+        implements GenericHomeLocal<T, PK>, GenericFacadeLocal<T, PK> {
```

Comment:
type parameters should comply naming conventions: https://docs.oracle.com/javase/tutorial/java/generics/types.html